src/importProvider.ts

```diff
@@ -1,213 +0,0 @@
-'use strict';
-import * as vscode from 'vscode';
-import * as Url from 'url';
-import { Uri, ViewColumn } from 'vscode';
-import * as fs from 'fs';
-import * as path from 'path';
-import { guessFileExtension } from './dataUriTextDocumentContentProvider';
-
-export async function load(sourceFilename: string) {
-    if (typeof sourceFilename == 'undefined') {
-        return;
-    }
-    if (!fs.existsSync(sourceFilename)) {
-        throw new Error('File not found.');
-    }
-
-    // Read the GLB data
-    const Binary = {
-        Magic: 0x46546C67
-    };
-
-    const sourceBuf = fs.readFileSync(sourceFilename);
-    const readMagic = sourceBuf.readUInt32LE(0);
-    if (readMagic !== Binary.Magic) {
-        throw new Error('Source file does not appear to be a GLB (glTF Binary) model.');
-    }
-    const readVersion = sourceBuf.readUInt32LE(4);
-    if (readVersion !== 2) {
-        throw new Error('Only GLB version 2 is supported for import. Detected version: ' + readVersion);
-    }
-
-    // Compose a target filename
-    let targetFilename = sourceFilename.replace('.glb', '.gltf');
-
-    if (!vscode.workspace.getConfiguration('glTF').get('alwaysOverwriteDefaultFilename')) {
-        const options: vscode.SaveDialogOptions = {
```

Comment:
I like that this code has been broken out to a reusable npm package now.  One UX wrinkle here though, I had moved this save file dialog to the middle of things so that we could ensure the loaded GLB was valid (correct version at least) before presenting the user with a save as dialog.  I didn't like seeing the save dialog first, and choosing an output name to save to, only to be told that the file I was loading wasn't even version 2.

Can the npm package have a "get output name" thenable here, so that it only requests the output name after it has verified the version number?

Reply:
Resolved